[BUG] Unused function warnings using `py_safe_call_object_once`

### Describe the bug

When I try to build a [branch of SciPy](https://github.com/scipy/scipy/compare/main...ngoldbaum:scipy:thread-safe-kdtree?expand=1) where I fix a thread-safety issue with `py_safe_call_object_once`, I get some compiler warnings about unused functions:

```
scipy/spatial/_ckdtree.cpython-314t-darwin.so.p/_ckdtree.cpp:3714:29: warning: function '__pyx_libcpp_mutex_limited_api_ensure_gil' is not needed and will not be emitted [-Wunneeded-internal-declaration]
 3714 |     static PyGILState_STATE __pyx_libcpp_mutex_limited_api_ensure_gil() {
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
scipy/spatial/_ckdtree.cpython-314t-darwin.so.p/_ckdtree.cpp:3732:16: warning: function '__pyx_libcpp_mutex_has_gil' is not needed and will not be emitted [-Wunneeded-internal-declaration]
 3732 |     static int __pyx_libcpp_mutex_has_gil() {
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~
scipy/spatial/_ckdtree.cpython-314t-darwin.so.p/_ckdtree.cpp:3852:17: warning: unused function '__pyx_libcpp_mutex_unlock' [-Wunused-function]
 3852 |     inline void __pyx_libcpp_mutex_unlock() {} // no-op
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~
3 warnings generated.
```

### Code to reproduce the behaviour:

Sorry, I don't have a small runnable example. I suspect all that's needed is cimporting and calling `py_safe_call_object_once`.


### Expected behaviour

No compiler warnings.

### OS

MacOS

### Python version

3.14.3t

### Cython version

3.2.4

### Additional context

_No response_

--- a/Cython/Includes/libcpp/mutex.pxd
+++ b/Cython/Includes/libcpp/mutex.pxd
@@ -158,7 +158,7 @@ cdef extern from *:
 
     namespace {
 
-    static PyGILState_STATE __pyx_libcpp_mutex_limited_api_ensure_gil() {
+    CYTHON_UNUSED PyGILState_STATE __pyx_libcpp_mutex_limited_api_ensure_gil() {
         #if CYTHON_COMPILING_IN_LIMITED_API
         if ((__PYX_LIMITED_VERSION_HEX < 0x030d0000) && __Pyx_get_runtime_version() < 0x030d0000) {
             return PyGILState_Ensure();
@@ -168,7 +168,7 @@ cdef extern from *:
     }
 
     #if CYTHON_COMPILING_IN_LIMITED_API && __PYX_LIMITED_VERSION_HEX < 0x030d0000
-    static void __pyx_libcpp_mutex_limited_api_release_gil(PyGILState_STATE gil_state) {
+    CYTHON_UNUSED void __pyx_libcpp_mutex_limited_api_release_gil(PyGILState_STATE gil_state) {
         if (__Pyx_get_runtime_version() < 0x030d0000)
             PyGILState_Release(gil_state);
     }
@@ -176,7 +176,7 @@ cdef extern from *:
     #define __pyx_libcpp_mutex_limited_api_release_gil(ignore) (void)ignore
     #endif
 
-    static int __pyx_libcpp_mutex_has_gil() {
+    CYTHON_UNUSED int __pyx_libcpp_mutex_has_gil() {
         #if CYTHON_COMPILING_IN_LIMITED_API
             if ((__PYX_LIMITED_VERSION_HEX >= 0x030d0000) || __Pyx_get_runtime_version() >= 0x030d0000) {
                 // In 3.13+ we can temporarily give up the GIL to find out what the thread state was
@@ -296,7 +296,7 @@ cdef extern from *:
         std::lock(arg0, arg1, args...);
     }
 
-    inline void __pyx_libcpp_mutex_unlock() {} // no-op
+    CYTHON_UNUSED inline void __pyx_libcpp_mutex_unlock() {} // no-op
 
     template <typename Lockable0T, typename ... Lockables>
     void __pyx_libcpp_mutex_unlock(Lockable0T& arg0, Lockables&... locks) {

